src/main/java/fr/techad/edc/httpd/utils/TokenUtils.java

```diff
@@ -0,0 +1,78 @@
+package fr.techad.edc.httpd.utils;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTCreationException;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+
+public class TokenUtils {
+	static final Logger LOGGER = LoggerFactory.getLogger(TokenUtils.class);
+	final String SECRET = "edc-server";
+	final String AUTH = "edc";
+
+	private static TokenUtils instance;
+	private String privateKey;
+	private FileUtils fileUtils;
+	private final String tokenPath = "./token.info";
+	private final String keyPath = "./private.key";
+
+	private TokenUtils() {
+		this.fileUtils = FileUtils.getInstance();
+		createTokenFile();
+	}
+
+	public static synchronized TokenUtils getInstance() {
+		if (instance == null)
+			instance = new TokenUtils();
+		return instance;
+	}
+
+	String genSecretKey() {
+		return RandomStringUtils.randomAlphanumeric(24);
+	}
+
+	public boolean validateToken(String token) {
+		if(readPrivateKey().isEmpty())return false;
```

Comment:
You don't solve the problem. You just check the key is contained in the file but this.privateKey can bu null